keanu-project/src/main/java/io/improbable/keanu/vertices/Vertex.java

```diff
@@ -262,4 +266,30 @@ public void saveValue(NetworkWriter netWriter) {
     public void loadValue(NetworkReader reader) {
        reader.loadValue(this);
     }
+
+    /**
+     * @return a DOT format label for this vertex, containing information such as vertex class name, display name and vertex label.
+     */
+    public VertexDotLabel getDotLabel() {
+        VertexDotLabel vertexDotLabel = new VertexDotLabel(this);
+        if (this.getLabel() != null) {
+            vertexDotLabel.setDotLabel(VertexDotLabel.VertexDotLabelType.VERTEX_LABEL, this.getLabel().getUnqualifiedName());
+        }
+        DisplayInformationForOutput vertexAnnotation = this.getClass().getAnnotation(DisplayInformationForOutput.class);
+        if (vertexAnnotation != null && !vertexAnnotation.displayName().isEmpty()) {
+            vertexDotLabel.setDotLabel(VertexDotLabel.VertexDotLabelType.ANNOTATION_LABEL, vertexAnnotation.displayName());
+        }
+        return vertexDotLabel;
+    }
+
+    /**
+     * @return a set of graph edges containing information about connected vertices.
+     */
+    public Set<GraphEdge> getParentEdgesInDotFormat() {
+        Set<GraphEdge> edges = new HashSet<>();
+        for (Vertex v : this.getParents()) {
+            edges.add(new GraphEdge(v, this));
+        }
+        return edges;
+    }
```

Comment:
Can this code live in the DotWriter class?  This would probably make more sense given the layout as that keeps all of the Dot specific code in one place and away from the Vertex class (which is what was requested for the protobuf stuff - obviously the only exception to that is any necessary annotations).

Reply:
Moved it to DotWriter now, I had to leave it in vertex previously to deal with hyperparameters (when they weren't annotated properly yet).